### ISSUE TITLE
Rails 4.2.x does not work with Ruby 2.4

### ISSUE THREAD

**Issue by ZipoKing** [rails/rails#27670]
### Steps to reproduce
1. Install Ruby 2.4
2. Try to launch e.g. Rails console

(Guidelines for creating a bug report are [available
here](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#creating-a-bug-report))

### Expected behavior
Rails console should be railed

### Actual behavior
SystemStack error is caused by activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb line 131

### System configuration
**Rails version**: 4.2.7.1

**Ruby version**: 2.4.0

```
$ /opt/ruby24/bin/rails c
/opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Fixnum is deprecated
/opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Bignum is deprecated
/opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:124:in `block (2 levels) in <class:Numeric>': stack level too deep (SystemStackError)
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/activesupport-4.2.7.1/lib/active_support/core_ext/numeric/conversions.rb:131:in `block (2 levels) in <class:Numeric>'
         ... 4062 levels...
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/railties-4.2.7.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
        from /opt/ruby24/lib/ruby/gems/2.4.0/gems/railties-4.2.7.1/lib/rails/commands.rb:17:in `<top (required)>'
        from bin/rails:4:in `require'
        from bin/rails:4:in `<main>'
```

**Comment by robin850** [comment]
Hi @ZipoKing,

Yes, Rails 4.2 isn't yet compatible with Ruby 2.4. This has been fixed in https://github.com/rails/rails/pull/27473 but you will need to wait for a new release of 4.2.x to be cut to be able to use it with Ruby 2.4.

You can still reference the GitHub repository in your Gemfile like:

~~~ruby
gem 'rails', github: 'rails/rails', branch: '4-2-stable'
~~~

I'm giving this a close but if the problem persists, feel free to comment so we can reopen that ! :-)

**Comment by mrzor** [comment]
If you land on this issue looking for the fix : Update to rails 4.2.8 or higher for ruby 2.4 support.
See https://github.com/jrafanie/parallel/commit/64eae43a1f00a9be6fc4017fc162803fd7467fdf
